### ISSUE TITLE
Pervasive Duplication in VSCode Extension

### ISSUE THREAD

**Issue by wenkokke** [haskell/haskell-language-server#4461]
### Your environment

_Which OS do you use?_
MacOS Ventura (13.7)

_Which version of GHC do you use and how did you install it?_
GHC versions from the 9.4, 9.6, and 9.8 series installed via ghcup.

How is your project built (alternative: link to the project)?
Various projects built using Cabal.

Which LSP client (editor/plugin) do you use?
VSCode and VSCodium with `haskell.haskell`.

Which version of HLS do you use and how did you install it?
HLS version 2.9.0.1 installed via ghcup.

Have you configured HLS in any way (especially: a `hie.yaml` file)?
Various projects with and without `hie.yaml` files.
Consistently configured to discover GHC and other tools via the PATH.

### Observed behaviour

The VSCode language extension for Haskell pervasively duplicates responses. This manifests in various ways:

- Duplicated Quick Fix suggestions.
- Overlapping Instance errors pointing to the same instance twice.
- ...

Unfortunately, the issue does not always happen. However, I can attest that it is not exclusively triggered by opening out-of-workspace files—which I don't generally do—or opening multiple workspaces concurrently—which I do often do.

### Steps to reproduce

I do not currently have a way to reproduce the behaviour, but can look into this once we hone in on a particular issue.

NOTE: I am opening this issue now, but I'll continue to add evidence as I gather it.

**Comment by fendor** [comment]
Thank you for opening this issue!

We additionally track this issue in https://github.com/haskell/vscode-haskell/issues/480

**Comment by wenkokke** [comment]
> Thank you for opening this issue!
> 
> We additionally track this issue in [haskell/vscode-haskell#480](https://github.com/haskell/vscode-haskell/issues/480)

Yes, I believe they're likely related. However, I didn't see my observations of duplicate suggestions and overlapping instance errors reproduced there, so I opened a new issue.

**Comment by fendor** [comment]
> Overlapping Instance errors pointing to the same instance twice.

That's a different issue #3694, I don't think that's part of the duplication issue.
